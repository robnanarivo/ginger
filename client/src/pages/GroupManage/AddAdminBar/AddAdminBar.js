import React, { useState } from 'react';
import { useParams } from 'react-router-dom';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import { updateGroup } from '../../../api';

export default function AddAdminBar(props) {
  const { users, updateAdmins, admins } = props;
  const { groupId } = useParams();
  const [selectedOption, setSelectedOption] = useState({ userName: '' });
  const adminsName = [];
  admins.forEach((admin) => { adminsName.push(admin.userName); });
  const handleChange = (event, value) => {
    setSelectedOption(value);
  };

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        width: '100%',
        backgroundColor: '#FFFFFF',
        maxWidth: 720,
        height: 90,
      }}
    >
      <Autocomplete
        freeSolo
        id="free-solo"
        sx={{ width: '600px', marginLeft: 3 }}
        disableClearable
        key={admins}
        options={users.filter((user) => !adminsName.includes(user.userName))}
        onChange={handleChange}
        getOptionLabel={(option) => option.userName}
        renderInput={(params) => (
          <TextField
            {...params}
            label="User Name"
            InputProps={{
              ...params.InputProps,
              type: 'search',
            }}
          />
        )}
      />
      <Button
        variant="contained"
        sx={{ marginLeft: 5, marginRight: 5, minWidth: '100px' }}
        onClick={async () => {
          if (selectedOption.id) {
            await updateGroup(groupId, 'join', [], [selectedOption.id]);
            await updateAdmins();
          }
        }}
      >
        ADD
      </Button>
    </Box>
  );
}
