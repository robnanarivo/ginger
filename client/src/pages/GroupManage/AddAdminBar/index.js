import AddAdminBar from './AddAdminBar';

export default AddAdminBar;
