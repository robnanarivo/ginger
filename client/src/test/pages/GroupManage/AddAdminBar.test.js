import React from 'react';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
// import { render, screen } from '@testing-library/react';
import AddAdminBar from '../../../pages/GroupManage/AddAdminBar/AddAdminBar';
import 'jest-styled-components';

describe('AddAdminBar unit test', () => {
  it('AddAdminBar Snapshot test', () => {
    const admins = [
      { id: 123, userName: 'goodboy' },
    ];

    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <Router history={history}>
          <AddAdminBar admins={admins} users={admins} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
