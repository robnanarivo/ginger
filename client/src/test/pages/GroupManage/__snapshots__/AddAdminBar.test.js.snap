// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAdminBar unit test AddAdminBar Snapshot test 1`] = `
<div
  className="MuiBox-root css-11fmhea"
>
  <div
    aria-expanded={false}
    aria-owns={null}
    className="MuiAutocomplete-root css-3ja3g8-MuiAutocomplete-root"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="free-solo"
        id="free-solo-label"
      >
        User Name
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-activedescendant={null}
          aria-autocomplete="list"
          aria-controls={null}
          aria-invalid={false}
          autoCapitalize="none"
          autoComplete="off"
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1j5luls-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="free-solo"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          required={false}
          spellCheck="false"
          type="search"
          value=""
        />
        <div
          className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-173wfxe"
          >
            <span>
              User Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-dcq3k7-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    ADD
  </button>
</div>
`;
